WindDocSync/sync.py

```diff
@@ -3,45 +3,61 @@
 from os.path import join, dirname
 from dotenv import load_dotenv
 import paho.mqtt.client as mqtt
+import re
+
+# convert the full card uid to a shorter format
+# compatible with wiegand output of esp-rfid
+def card_number_to_wiegand_format(card_number: str) -> str:
+  # make all cards 14 bytes long
+  hex_string_pad = card_number.zfill(14)
+  # keep only the first 8 bytes
+  hex_string_cut = hex_string_pad[:8]
+  hex_data = bytearray.fromhex(hex_string_cut)
+  hex_data.reverse()
+  return hex_data.hex()
+
+def load_users():
+  sync_json_path = join(dirname(__file__), 'sync.json')
+  file = open(sync_json_path)
+  users = json.load(file)
+  os.remove(sync_json_path)
+  return users
+
+def mqtt_delete_users():
+  mqttClient.publish('esp-rfid/cmd', json.dumps({
+     'cmd': 'deletusers',
+     'doorip': ESPRFID_IP
+  }))
+
+def mqtt_add_users(users):
+  for u in users:
+    pincode = u['Pin']
+    # default pincode if the provided is not valid
+    # valid pincode is made by 4 digits
+    if re.match(r'\d{4}', pincode) is None:
+      pincode = '0000'
```

Comment:
I propose to bring this to xxxx (not typeable on the reader) so as to force them to insert a custom pin. 0000 is too common as a pin and would not be customized

Reply:
it is a double check because it is already done in sync.php